### ISSUE TITLE
Document the semantics of batch.

### ISSUE THREAD

**Issue by Stebalien** [ipfs/go-datastore#102]
* Can it flush early?
* How is it different from a transaction?

Currently, it's *entirely* undocumented.

**Comment by Stebalien** [comment]
@bigs @magik6k thoughts?

**Comment by bigs** [comment]
i'd push to have it's semantics not be transactional, i.e. writes in a `Batch` are not guaranteed to happen in the same iota of time. i'd say early flush is a no?

**Comment by Stebalien** [comment]
Yeah, those both sound like good ideas. Transactions *are* batches but batches aren't transactions (and can partially fail). Would you have time to document this (on the interface definition)?

**Comment by bigs** [comment]
absolutely!

**Comment by raulk** [comment]
We can also characterise them in terms of properties. Batches provide no ACID guarantees and are mainly used for performance reasons; transactions provide atomicity and isolation guarantees.
